tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -105,7 +107,61 @@ propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full
             // modify the command to add the comment
             var commandText = command.CommandText ?? string.Empty;
             var propagationComment = StringBuilderCache.GetStringAndRelease(propagatorStringBuilder);
-            if (ShouldAppend(integrationId, commandText))
+            if (command.CommandType == CommandType.StoredProcedure && integrationId == IntegrationId.SqlClient) // only supported for Microsoft SQL Server
+            {
+                // Save the original stored procedure name
+                string procName = command.CommandText ?? string.Empty;
+
+                if (string.IsNullOrEmpty(procName))
+                {
+                    return false;
+                }
+
+                // Build parameter list for EXEC statement
+                StringBuilder paramList = new StringBuilder();
+                foreach (DbParameter? param in command.Parameters)
+                {
+                    if (param == null)
+                    {
+                        continue;
+                    }
+
+                    // Skip return value parameters
+                    if (param.Direction == ParameterDirection.ReturnValue)
+                    {
+                        continue;
+                    }
+
+                    if (paramList.Length > 0)
+                    {
+                        paramList.Append(", ");
+                    }
+
+                    paramList.Append(param.ParameterName).Append('=').Append(param.ParameterName);
+
+                    if (param.Direction == ParameterDirection.InputOutput || param.Direction == ParameterDirection.Output)
+                    {
+                        // For OUTPUT parameters, we need to add the OUTPUT keyword
+                        paramList.Append(" OUTPUT");
+                    }
+                }
+
+                // Change command type to Text
+                command.CommandType = CommandType.Text;
```

Comment:
This causes several things to happen:

`SqlCommand.cs` will treat this as a normal  query and build it up via [this function as an example](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L5617)

Compared to [BuildRpc](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System.Data/System/Data/SqlClient/SqlCommand.cs#L5548)

This `BuildRpc` shows why the initial `CommandText` injection doesn't work (as it just uses that as the RPC name directly)

Reply:
Added a comment at the top of this statement that explains more.